engines/dfc_provider/app/services/affiliate_sales_data_builder.rb

```diff
@@ -0,0 +1,171 @@
+      DataFoodConsortium::Connector::Enterprise.new(urls.enterprise_url(id))
+    end
+
+    def build_address(sale)
+      DataFoodConsortium::Connector::Address.new(
+        urls.address_url(sale.producer_id),
+        postalCode: sale.producer_postcode,
+        country: nil,
+        latitude: nil,
+        longitude: nil,
+        region: nil,
+        street: nil,
+        city: sale.producer_city
+      )
+    end
+
+    def build_producer(sale)
+      DataFoodConsortium::Connector::Enterprise.new(
+        urls.enterprise_url(sale.producer_id),
+        logo: nil,
+        name: nil,
+        description: nil,
+        vatNumber: nil,
+        suppliedProducts: build_supplied_product(sale),
+        localizations: build_address(sale)
+      )
+    end
+
+    def build_supplied_product(sale)
+      DataFoodConsortium::Connector::SuppliedProduct.new(
+        urls.enterprise_supplied_product_url(sale.producer_id, sale.product_id),
+        name: sale.product_name,
+        description: nil,
+        alcoholPercentage: nil,
+        lifetime: nil,
+        usageOrStorageConditions: nil,
+        totalTheoreticalStock: nil,
+        catalogItems: build_catalogue_item(sale),
+        quantity: build_quantity(sale)
+      )
+    end
+
+    def build_catalogue_item(sale)
+      DataFoodConsortium::Connector::CatalogItem.new(
+        urls.enterprise_catalog_item_url(sale.producer_id, sale.product_id),
+        stockLimitation: nil,
+        sku: nil,
+        offers: build_offer(sale)
+      )
+    end
+
+    def build_offer(sale)
+      DataFoodConsortium::Connector::Offer.new(
+        urls.enterprise_offer_url(sale.producer_id, sale.product_id),
+        stockLimitation: nil,
+      )
+    end
+
+    def build_order_line(sale)
+      DataFoodConsortium::Connector::OrderLine.new(
+        urls.enterprise_order_order_line_url(sale.producer_id, sale.order_id, sale.line_item_id),
+        description: nil,
+        offer: build_offer(sale),
+        order: build_order(sale),
+        quantity: build_quantity(sale),
+        price: build_price(sale)
+      )
+    end
+
+    def build_order(sale)
+      DataFoodConsortium::Connector::Order.new(
+        urls.enterprise_order_url(sale.producer_id, sale.order_id),
+        number: nil,
+        date: sale.order_date.strftime("%Y-%m-%d")
+        # saleSession: build_sale_session(sale)
+      )
+    end
+
+    def build_sale_session(sale)
+      DataFoodConsortium::Connector::SaleSession.new(
+        urls.enterprise_sale_session_url(sale.producer_id, sale.line_item_id),
+        beginDate: nil,
+        endDate: nil,
+        quantity: nil
+      )
+    end
+
+    def build_quantity(sale)
+      DataFoodConsortium::Connector::Quantity.new(
+        unit: sale.unit_presentation,
+        value: sale.line_item_quantity
+      )
+    end
+
+    def build_price(sale)
+      DataFoodConsortium::Connector::Price.new(
+        value: sale.price.to_f,
+        unit: sale.currency
```

Comment:
This should also be a constant like `connector.MEASURES.EURO`. Have a look at https://github.com/datafoodconsortium/connector-ruby/.